src/client/runner/runner.js

```diff
@@ -121,8 +130,16 @@ Runner.prototype._reportErrorToServer = function (err, isAssertion) {
     return new Promise(resolve => {
         if (isAssertion)
             transport.assertionFailed(err, resolve);
-        else
+        else if (beforeUnloadRaised) {
+            // NOTE: we should not stop the test run if an error occured during page unloading because we
+            // would destroy the session in this case and wouldn't be able to get the next page in the browser.
+            // We should set the deferred error to the task to have the test fail after the page reloading.
+            this.testContextStorage.get().deferredError = err;
```

Comment:
we set the `deferredError` here but try to get `testError` [here](https://github.com/DevExpress/testcafe/pull/313/files#diff-619aff4845280917d61f0dfe5e1a6071R39)